course/src/vector3.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * Isometry library
+ * Copyright 2020 Ekumen Inc.
+ * Author: Alexis Pojomovsky, 2020
+ */
+
+#include <isometry/vector3.hpp>
+#include <stdexcept>
+
+namespace ekumen {
+namespace math {
+
+const Vector3 Vector3::kZero{Vector3(0., 0., 0.)};
+const Vector3 Vector3::kUnitX{Vector3(1., 0., 0.)};
+const Vector3 Vector3::kUnitY{Vector3(0., 1., 0.)};
+const Vector3 Vector3::kUnitZ{Vector3(0., 0., 1.)};
+
+Vector3::Vector3() {
+  x_ = 0.;
+  y_ = 0.;
+  z_ = 0.;
+}
+
+Vector3::Vector3(double _x, double _y, double _z) : x_{_x}, y_{_y}, z_{_z} {}
+
+Vector3 Vector3::operator+(const Vector3& vector) const {
+  return Vector3{x() + vector.x(), y() + vector.y(), z() + vector.z()};
```

Comment:
Avoid duplicating the implementation of the operators. You have an implementation for `+=`, so you can use that in `+`.

```
Vector3 Vector3::operator+(const Vector3& vector) const {
  Vector3 aux{*this};
  aux += vector;
  return aux;
}
```

Reply:
Same in other operator pairs below.